onnxruntime/test/optimizer/qdq_test_utils.cc

```diff
@@ -98,5 +98,39 @@ GetQDQTestCaseFn BuildQDQConcatTestCase(const std::vector<std::vector<int64_t>>&
   };
 }
 
+/*
+#ifdef USE_NNAPI
+GetQDQTestCaseFn BuildQDQConcatTestCaseUnsupported() {
+  return [](ModelTestBuilder& builder) {
+    const std::vector<std::vector<int64_t>>& input_shapes = {
+        {1, 6, 36},
+        {1, 6, 8},
+        {1, 6, 2},
+    };
+    int64_t axis = 2;
+
+    std::vector<NodeArg*> input_args;
+    std::vector<NodeArg*> q_input_args;
+    
+    // set unmatched input scales/zp for test purpose
+    input_args.push_back(builder.MakeInput<float>(input_shapes[0], -1.f, 1.f));
+    q_input_args.push_back(AddQDQNodePair<uint8_t>(builder, input_args.back(), 0.05f, 128));
+    input_args.push_back(builder.MakeInput<float>(input_shapes[1], -1.f, 1.f));
+    q_input_args.push_back(AddQDQNodePair<uint8_t>(builder, input_args.back(), 0.04f, 127));
+    input_args.push_back(builder.MakeInput<float>(input_shapes[2], -1.f, 1.f));
+    q_input_args.push_back(AddQDQNodePair<uint8_t>(builder, input_args.back(), 0.03f, 126));
```

Comment:
For QDQConcat, when we make the inputs scale/zp unmatched, supposedly I guess it shouldn't be supported by NNAPI.
but TestCase TestQDQConcat_UnsupportedInputScalesAndZp (commented out) is failing, showing there are nodes can handled by NNAPI. not sure about this..

Reply:
NNAPI only requires concat inputs/output has same scales and zps in API 28-, our CI runs on API 29, so this won't fail
This can only be tested by forcing API 28 as the `OpSupportCheckParams::android_feature_level` for this function here
https://github.com/microsoft/onnxruntime/blob/1e4a4bfe58de18d07a66a458ff664cb3d998cdcb/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.h#L147

Reply:
Could we set/read the level using an environment variable for non-android builds? i.e. add a check for the env var [here](https://github.com/microsoft/onnxruntime/blob/4ef81b142dbd29abc3c64797084b8e61cf3212d1/onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc#L95)